kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue

```diff
@@ -193,8 +193,19 @@
               routeBase = '/topics/t';
               path = `${routeBase}/${id}`;
               router.push({ path: path }).catch(() => {});
-            } else if (contentNode) {
-              // in a custom context, launch or maintain overlay
+            } else if (contentNode && this.overlayIsOpen == false) {
+              // in a custom context, launch overlay
+              this.currentContent = contentNode;
+              this.overlayIsOpen = true;
+              context.node_id = contentNode.id;
+              context.customChannel = true;
+              const encodedContext = encodeURI(JSON.stringify(context));
+              router.replace({ query: { context: encodedContext } }).catch(() => {});
+            } else if (contentNode && this.overlayIsOpen == true) {
+              // in a custom context, within an overlay, switch the overlay
+              // content to the new content
+              this.overlayIsOpen = false;
```

Comment:
Given that you are not waiting for `$nextTick` here, I'm not sure that setting false and then true on `overlayIsOpen` is doing anything. Better off ensuring the overlay renderer is properly reactive to changing its content node, which I think it probably is.